### PR TITLE
port https://github.com/tektoncd/pipeline/pull/2967 to triggers

### PR DESCRIPTION
**Pull request by gabemontero**
# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
    The PodSecurityPolicy configuration was updated so that container must run as non-root, and the RBAC for utilizing the PSP was moved from cluster scoped to namespace scoped to better ensure triggers can not utilize other PSPs unrelated to this project
```

### CODE REVIEW

**Comment by gabemontero** [comment]
@vdemeester @nikhil-thomas @sthaha FYI

IIRC the analogous upstream pipeline PR that modified PSP/roles had implications for us downstream ?

File: config/200-role.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: tekton-triggers-admin
+  labels:
+    app.kubernetes.io/instance: default
+    app.kubernetes.io/part-of: tekton-triggers
+rules:
+  - apiGroups: ["policy"]
+    resources: ["podsecuritypolicies"]
+    resourceNames: ["tekton-pipelines"]
```
Comment (dibyom) [inline config/200-role.yaml:25]: should be tekton-triggers

File: config/200-role.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Copyright 2019 The Tekton Authors
```
Comment (savitaashture) [inline config/200-role.yaml:1]:
```suggestion
# Copyright 2020 The Tekton Authors
```

File: config/201-rolebinding.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Copyright 2019 The Tekton Authors
```
Comment (savitaashture) [inline config/201-rolebinding.yaml:1]:
```suggestion
# Copyright 2020 The Tekton Authors
```

**Comment by gabemontero** [comment]
@dibyom @savitaashture I finally pushed updates based on your comments last month

I've also reached out to @skaegi about verifications in his env's vs. the verifications I'll be doing.

**Comment by gabemontero** [comment]
I'm going to remove the WIP label ... I've completed the testing in the secured environments I have access to

**Comment by dibyom** [comment]
Could we drop the WIP from the commit message and make it a bit more descriptive? (also, add to the release notes?)

**Comment by savitaashture** [comment]
Hi @gabemontero 

I tried to install trigger using this PR on minikube and i see pods's are not coming up

```
kubectl get pods -n tekton-pipelines  -w
NAME                                           READY   STATUS             RESTARTS   AGE
tekton-triggers-controller-5995c76b5b-bgkfb    0/1     CrashLoopBackOff   5          10m
tekton-triggers-webhook-5c94c6cdc-cwmpx        0/1     CrashLoopBackOff   6          10m
```
with below error
```
ae8bdf6a3207335c"
  Warning  BackOff    100s (x5 over 2m26s)   kubelet, minikube  Back-off restarting failed container
  Normal   Created    87s (x5 over 3m6s)     kubelet, minikube  Created container tekton-triggers-controller
  Normal   Pulled     87s (x4 over 2m54s)    kubelet, minikube  Container image "docker.io/savita3020/controller-f656ca31de179ab913fa76abc255c315@sha256:0d2d80595fdfa8ee8462532fe03d04447dc6d35fc35c2e93ae8bdf6a3207335c" already present on machine
  Warning  Failed     86s (x5 over 3m6s)     kubelet, minikube  Error: failed to start container "tekton-triggers-controller": Error response from daemon: OCI runtime create failed: container_linux.go:345: starting container process caused "chdir to cwd (\"/home/nonroot\") set in config.json failed: permission denied": unknown
```

same issue occurring for pipeline https://github.com/tektoncd/pipeline/issues/3273

**Comment by gabemontero** [comment]
Ah ha ... thanks for trying that out @savitaashture 

Am I correct that you had to explicitly turn on pod security policy controller with your minikube installation (as I recall PSP controller is off by default with k8s ... or does minikube turn in on automatically under the covers) ?

I agree that it looks like the same situation in pipelines.

/hold

let's see how https://github.com/tektoncd/pipeline/issues/3273 sorts out

**Comment by gabemontero** [comment]
/hold cancel

I've pulled in https://github.com/tektoncd/pipeline/pull/3342 which addressed @savitaashture 's issue

/assign @dibyom 
/assign @savitaashture 

PTAL ... thanks

**Comment by gabemontero** [comment]
check that ... @dibyom pulled in that latest fix with 09539a8a3e956da4b452c82a697a8e1373c95e24

**Comment by dibyom** [comment]
This LGTM. One minor nit - can we make the commit message more descriptive? maybe, it could be the same as the message from the pipelines PR. Also, let's add to the release notes?

**Comment by gabemontero** [comment]
OK cleaned up commit message and got the rebase sorted out

PTAL @dibyom and @savitaashture

**Comment by gabemontero** [comment]
release note updated as well @dibyom

**Pull request merged**